File: Solutions/1207. Unique Number of Occurrences/1207.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        HashMap<Integer, Integer> hm = new HashMap<>();
        for(int i: arr){
            if(hm.containsKey(i)){
                hm.put(i, hm.get(i) + 1);
            }else {
                hm.put(i,1);
            }
        }
        HashSet<Integer> hs = new HashSet<>();
        for(Map.Entry<Integer,Integer> map: hm.entrySet()){
            if(hs.contains(map.getValue())){
                return false;
            }
            hs.add(map.getValue());
        }
        return true;
    }
}

File: Solutions/1207. Unique Number of Occurrences/1207.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        map = {}
        for i in arr:
            if i in map:
                map[i] += 1
            else:
                map[i] = 1
        s = set()
        for i,j in map.items():
            if j in s:
                return False
            s.add(j)
        return True
        